src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -294,24 +294,27 @@ public virtual VerifyResult VerifyForEachBlock(StoreView snapshot, TransactionVe
             foreach (TransactionAttribute attribute in Attributes)
                 if (!attribute.Verify(snapshot, this))
                     return VerifyResult.Invalid;
-            if (hashes.Length != Witnesses.Length) return VerifyResult.Invalid;
-            for (int i = 0; i < hashes.Length; i++)
```

Comment:
This loop can't be removed.

Reply:
This loop is in func VerifyWitnesses in latter part. I split witness verification to 2 parts, independent and dependent with snapshot. The former will be checked in RemoteNode, and the latter will be checked here, including cases like contract invoking, etc.